### ISSUE TITLE
Mob Entity Does Not Work?

### ISSUE THREAD

**Issue by syeare** [PrismarineJS/mineflayer#1165]
For some reason, the bots don't attack any nearby entities. It seems they can detect them, though. They connect to the server and just straight up _attack_ when there are mobs nearby, but they don't target the actual mobs themselves. You could probably tell from looking through my garbage code, but I am somewhat new to JS. I used some of the snippets from the `jumper.js` example after I failed to implement what I needed to from scratch. I don't understand why exactly it doesn't work properly.

### Info
 - **mineflayer**: 2.24.0
 - **server**: paper 1.16.1
 - **node**: 12.18.2
 - **code**: 
https://gist.github.com/syeare/12ce077a4df18ae0a1fa216bf9255b9e

**Comment by rom1504** [comment]
The problem is the while. You are trying to attack the nearest entity every
tick.
Just repeat the attack every n seconds with a setInterval

**Comment by syeare** [comment]
You are referring to the while loop on line 27, right? I used an if conditional there originally, but it would just attack once on logging in/spawning and then just stand there. I updated the gist with what I did using `setInterval`, but it still doesn't seem to work. Additionally, there seems to be some kind of type error?

Error: https://gist.github.com/syeare/9cbc0c9d59f551a2c8a3e09f44055ce2

**Comment by rom1504** [comment]
yeah apparently you used setInterval wrong, what is your new code ?

**Comment by syeare** [comment]
I updated that same Gist @ https://gist.github.com/syeare/12ce077a4df18ae0a1fa216bf9255b9e

**Comment by rom1504** [comment]
setInterval takes a function as first argument, for example `() => bot.attack(entity, true)`
Maybe what you want to put in that function would also include the `entity = nearestEntity(bot);` so that entity get refreshed ?

**Comment by syeare** [comment]
Oh thanks, I updated the Gist again to reflect the changes. Although, I still don't understand why exactly the bots just stand there while getting attacked by zombies lmao

I changed `setInterval()` to the following:
```
setInterval(() => { entity = nearestEntity(bot); bot.attack(entity, true); }, 750);
```

**Comment by rom1504** [comment]
You need to remove the if entity or it doesn't execute the set interval at all

**Comment by syeare** [comment]
I did remove that if statement, but they still just stand there attacking in the direction they face when spawning. They do still activate whenever there is an entity, but they just don't look at it. I modified the `nearestEntity()` function from jumper.js. The main reason I wanted to make this bot is so that I can simulate real player actions to stress test just how much my server will lag with larger player counts
